### ISSUE TITLE
Slower than `node-glob` when ignoring a directory with `dir/**`

### ISSUE THREAD

**Issue by pvdlg** [mrmlnc/fast-glob#45]
See benchmark file: https://gist.github.com/pvdlg/8f72824d81263efc71b30fd443aa8b0e

```bash
npm install matcha, fast-glob, glob
./node_modules/.bin/matcha fast-blog-bench.js
```
##### patterns: `'a/*'`, ignore: `['a']`
231 op/s » fast-glob sync
136 op/s » glob sync

##### patterns: `'a/*'`, ignore: `['a/**']`
302 op/s » fast-glob sync
31,096 op/s » glob sync

It's quite an edge case as it seems the performance difference happens only when the the `ignore` pattern make so that no pattern will match. It seems that `node-glob` is able to figure out that nothing will ever with pattern `'a/*'` and ignore `['a/**']`, while `fast-glob` still goes through the directory.

With pattern `'a/*'` and ignore `['a']` it seems neither `fast-glob` nor `node-glob` figure out that nothing would match and both attempt to go through the directory.
I thought the directory would be ignored completely in such case.

I'm not sure it's a big issue as it concerns a pattern/ignore config that doesn't make much sense, but I though it worth mentioning in case the perf improvement is really simple to do. Maybe ignoring patterns that are included within ignore pattern can bring other more useful performance improvements.

**Comment by mrmlnc** [comment]
Hello, @pvdlg,

Thanks for good catch. Yeap, I know about this problem. The current behavior is described in [«How to exclude directory from reading?»](https://github.com/mrmlnc/fast-glob#how-to-exclude-directory-from-reading) section. 

Unfortunately, I don't remember why I decided to only describe the current behavior. It seems that were borderline cases when the directory still need to read the case even if you specify a pattern that excludes this directory. I will try to find ~artifacts~ notes of these conversations.

**Comment by mrmlnc** [comment]
```
test
├── a
│   ├── a.css
│   ├── a.js
│   └── a.txt
└── b
    ├── b.css
    ├── b.js
    └── b.txt
```

```js
const globby = require('globby');

const entries = globby.sync(['**/*', '!a/**', 'a/*.{js,css}'], { cwd: 'test' });

console.dir(entries, { colors: true });
```

In this case, `globby respects the order of patterns and results will be:

```
[ 'b/b.css', 'b/b.js', 'b/b.txt', 'a/a.css', 'a/a.js' ]
```

So in this case we need to read the `a` directory.

But the `fast-glob` package does not respect the order of patterns. First, all the negative patterns are applied, and only then the positive patterns. Results will be:

```
[ 'b/b.css', 'b/b.js', 'b/b.txt' ]
```

In this case, we can safely ignore the reading of directories that end with the globstar. I file PR #49 for this case.

**Comment by mrmlnc** [comment]
Landed by `fast-glob@2.0.3`.
